crypto/crypto.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+const encryptionPrefix = "aes-256:"
+
+var (
+	encryptionKey         = ""
+	encryptionFallbackKey = ""
+
+	ErrDecryptionFailed   = errors.New("unable to decrypt the data")
+	ErrCipherTextTooShort = errors.New("cipher text is too short")
+)
+
+func SetAESEncryptionKey(key string) {
+	encryptionKey = key
+}
+
+func SetAESEncryptionFallbackKey(key string) {
+	encryptionFallbackKey = key
+}
+
+func AESEncrypt(data string) ([]byte, error) {
+	return encrypt(data, encryptionKey)
+}
+
+func AESDecrypt(data []byte) (string, error) {
+	out, err := decrypt(data, encryptionKey)
+	if err == ErrDecryptionFailed {
+		out, err = decrypt(data, encryptionFallbackKey)
+	}
+	return out, err
+}
+
+func encrypt(data string, key string) ([]byte, error) {
+	if len(key) == 0 {
+		return []byte(data), nil
```

Comment:
Shouldn't we return error here?

Reply:
The idea here was `empty key = no encryption`

Reply:
What happens in case of prod misconfiguration? We won't encrypt credentials then; we can use CLI to reencrypt, but what for? Shouldn't we force using the encryption key? I'm not sure if this is a good idea, not encrypting credentials can stay hidden for a long time.

As agreed with Fabio, we'll discuss it during the next backend team meeting.